p2p/discover/table.go

```diff
@@ -337,7 +341,17 @@ func (tab *Table) doRevalidate(done chan<- struct{}) {
 	}
 
 	// Ping the selected node and wait for a pong.
-	err := tab.net.ping(unwrapNode(last))
+	remoteSeq, err := tab.net.ping(unwrapNode(last))
+
+	// Also fetch record if the node replied and returned a higher sequence number.
+	if last.Seq() < remoteSeq {
+		n, err := tab.net.requestENR(unwrapNode(last))
```

Comment:
Probably a good idea at this point to check that the returned version number is actually higher than last.Seq, as I don't think requestENR verifies that.

Reply:
Check added (to requestENR)